CRM/ManualDirectDebit/Hook/Custom/CancellationBatchChecker.php

```diff
@@ -0,0 +1,161 @@
+<?php
+use CRM_ManualDirectDebit_Common_MandateStorageManager as MandateStorageManager;
+
+/**
+ * Class CRM_ManualDirectDebit_Hook_Custom_CancellationBatchChecker.
+ *
+ * Checks if the mandate's status has changed vs active cancellations, to remove
+ * it from the batch if need be. MAndates that change their status from '0C' to
+ * anything else should be removed from pending (unsubmitted) batches.
+ */
+class CRM_ManualDirectDebit_Hook_Custom_CancellationBatchChecker {
+
+  private $contactID;
+  private $params;
+  private $mandateStorageManager;
+  private $dd_codes;
+
+  /**
+   * CRM_ManualDirectDebit_Hook_Custom_CancellationBatchChecker constructor.
+   *
+   * @param $contactID
+   * @param $params
+   * @param \CRM_ManualDirectDebit_Common_MandateStorageManager $mandateStorageManager
+   */
+  public function __construct($contactID, $params, MandateStorageManager $mandateStorageManager) {
+    $this->contactID = $contactID;
+    $this->params = $params;
+    $this->mandateStorageManager = $mandateStorageManager;
```

Comment:
The good thing of injecting it instead of instantiating here is that to create a test, we can use a mock MandateStorageManager object we control, so there would be no dependency on CiviCRM nor its database. We can just have the mock object return whatever we need it to return, in order to run the test.